google/cloud/storage/internal/object_streambuf.cc

```diff
@@ -68,9 +68,10 @@ StatusOr<ObjectReadStreambuf::int_type> ObjectReadStreambuf::Peek() {
   // assert(n <= current_ios_buffer_.size())
   current_ios_buffer_.resize(read_result->bytes_received);
 
-  for (auto const& kv : read_result->response.headers) {
+  auto hint = headers_.end();
+  for (auto& kv : read_result->response.headers) {
     hash_validator_->ProcessHeader(kv.first, kv.second);
-    headers_.emplace(kv.first, kv.second);
+    hint = headers_.emplace_hint(hint, kv.first, std::move(kv.second));
```

Comment:
How come you can't move `kv.first` too?

Reply:
It is a `std::string const`, because `value_type` for associative containers is `std::pair<K const, V>` as you probably know better than I do.

Reply:
Apparently not.